### ISSUE TITLE
Write ChordSelector

### ISSUE THREAD

**Issue by arr28** [DodgyJammers/jammypiece#10]
Heart of the product.  How do we choose sensible chords?

**Comment by arr28** [comment]
What chord progressions sound good?  General wisdom seems to say that you can go from I to anywhere, but should then follow links on a "good progression" map.

For example, from http://chordmaps.com/genmap.htm...

![image](https://cloud.githubusercontent.com/assets/3194913/7503906/de30623a-f43d-11e4-815d-025da640f43d.png)

(There are more sites like this.  This map seems more comprehensive than most.)

If using a probabilistic system, do the edges need weights?  Also, a desire to stay away from home initially but to return there later?  We probably care about phrase lengths.

**Comment by rgefryer** [comment]
Absolutely.  Feel free to have a play with that app you suggested on my iPad.

At the moment I'm thinking that we start with a home key, then move between chords based on:
- favoured progressions
- needing to return to the home key by a certain time (depending on "song" structure)
- notes coming out of the soloist
- intensity of the soloist (volume, note density, register)
- a pinch of randomness.

**Comment by kw217** [comment]
Whoever writes the Chord class for the chord generator (@CJR501?) should use the Note class I'm building as part of #22.

**Comment by arr28** [comment]
A simpler map is available [here](http://mugglinworks.com/chordmaps/part3.htm).  This is what's implemented by ChordMap right now.
